src/oncall/notifier/reminder.py

```diff
@@ -48,8 +48,12 @@ def reminder(config):
     cursor.execute('SELECT `last_window_end` FROM `notifier_state`')
     if cursor.rowcount != 1:
         window_start = int(time.time() - interval)
-        logger.warning('Corrupted/missing notifier state; unable to determine last window. Guessing %s',
+        logger.warning('Corrupted/missing notifier state; unable to determine last window. Guessing %s. Creating state in database',
                        window_start)
+        # create a clean state in the datebase
+        cursor.execute('TRUNCATE `notifier_state`')    
```

Comment:
Rather than truncating the table, why not retain the insert, and alter it to use the "on duplicate key update" syntax? This will work because the table has a primary key. Eg:

```
cursor.execute('INSERT INTO `notifier_state` (`last_window_end`) VALUES (%s) ON DUPLICATE KEY UPDATE `last_window_end` = values(last_window_end)', window_start)
```

In general, it's most common to reserve `truncate` queries to manual database maintenance rather than in general app flow.

Reply:
@jrgp  I've replaced `TRUNCATE` with `DELETE`.